app/controllers/components_controller.rb

```diff
@@ -2,15 +2,19 @@
 
 class ComponentsController < ApplicationController
   def show
-    @component = Component.get(params[:id])
+    @component = Component.get(params[:id]) || not_found
   end
 
   def index
     @components = Component.all
   end
 
   def preview
-    @component = Component.get(params[:component_id])
+    @component = Component.get(params[:component_id]) || not_found
     render layout: 'preview'
   end
+
+  def not_found
+    render status: 404, text: "404 error"
```

Comment:
If I do that then the line seems to have no effect and I get the original exception that I was trying to prevent.